addons/account/models/account_move.py

```diff
@@ -5280,8 +5280,10 @@ def _find_and_set_purchase_orders(self, po_references, partner_id, amount_total,
 
     def _link_bill_origin_to_purchase_orders(self, timeout=10):
         for move in self.filtered(lambda m: m.move_type in self.get_purchase_types()):
-            references = [move.invoice_origin] if move.invoice_origin else []
+            references = move.invoice_origin.split() if move.invoice_origin else []
             move._find_and_set_purchase_orders(references, move.partner_id.id, move.amount_total, timeout=timeout)
```

Comment:
i suppose it wasn't working before when multiple reference where given, maybe a test would be nice for that case